Add additional data to OpenSearch index.

ItIsJordan opened the pull request:
Closes #557.  Adds the `additional_resources` of a submission object, and the `data_abstract` (`comment` from `submission.yaml`) to the OpenSearch index as mentioned in #557. Also updates a number of tests which broke upon adding new test data.

ItIsJordan committed:
Add data_abstract to opensearch index

Adds the data_abstract (comment in yaml) to the opensearch indexer function.

ItIsJordan committed:
Fix test_submissions_csv in dashboard_test.py

Fixes test_submissions_csv in dashboard_test.py to use regex when checking against URLs in text as the URL on the webpage does not always match what is contained in the Flask config.

ItIsJordan committed:
Add data abstract search test

Adds testing for data abstract searching

ItIsJordan committed:
Revert "Fix test_submissions_csv in dashboard_test.py"

This reverts commit 5c67dc8eda88cc93b04d9b7481fa2f043858423c.

ItIsJordan committed:
Add table data description to opensearch index

Adds table data to the OpenSearch index for submission searching in the OpenSearch document enhancers.

ItIsJordan committed:
Update search testing to include table description searching

Updates test_search in search_test.py to include testing of table description text

ItIsJordan committed:
Revert "Add table data description to opensearch index"

This reverts commit 4cf908a4b491e395a15dae4a0aed11e274e53c91.

ItIsJordan committed:
Add additional resource descriptions to index

Updates index to add file_description from additional resource objects within HEPSubmission/DataSubmission objects to the OpenSearch index.

ItIsJordan committed:
Merge branch 'main' into add-additional-desc-to-osindex

ItIsJordan committed:
Update test for resource searching in OpenSearch API

Updates search_test OpenSearch testing to properly include some testing for searching of additional resources.

ItIsJordan committed:
Update hard-coded/assumed testing variables

Updates the hard-coded/assumed variables used as expected data within the tests to accommodate the existence of a new test submission in the identifiers dictionary within conftest.py

ItIsJordan committed:
Update broken test in test_dashboard.py

Updates a broken test in test_dashboard.py, this test has issues running on my local system.

ItIsJordan committed:
Fix unrendered LaTeX in table description

Now triggers rendering of LaTeX in record table descriptions as this was bugged and only occurred when the full table is loaded. Mentioned in issue #756

ItIsJordan committed:
Clear table quantity text on records page

Fixes a bug where the table_options_region element ("Showing X of X values" and "Show All") on the records page would sometimes have the data of another table if loaded later.

ItIsJordan committed:
Refactor data table routes

Refactors the route to get data tables at /data/ to separate the data table (yaml file) and the headers query code. Implements a second route to select only the table data.

ItIsJordan committed:
Update table renderer js

Updates the table renderer JS to properly use new route to select only data table.

ItIsJordan committed:
Update records_test.py

Updates tests in response to previous commit.

ItIsJordan committed:
Fix button query bug

Fixes an issue as a part of the large file loading bugs in #756 where the filesize threshold loading button would execute multiple queries. This occurred by multiple events being passed to the button. Now clears the buton events before.

ItIsJordan committed:
Revert large file bugfixes to move to new branch (last 6 commits)

Reverts a number of bugfixes/changes for the large file loading functionality that was intended for a separate branch.

Reverts commits:

12214c7adabc0e19223620c75366e5a2c2b4a748
4afcb2fd7eb9a95157acccdb1ce4f2f550270285
a9bd63b75ad8723a3a96421321dcad1456258407
4dc662a9d97f34848c3986a984cf1761e9621091
6445bd97b36c8a4ced1d1dea44ac10b3a1ccba33
882d6a92d70e5b82ae077803a89871ee38267b79

ItIsJordan committed:
Merge branch 'main' into add-additional-desc-to-osindex

GraemeWatt reviewed (changes requested):
Thanks, this is great!  I tested and it works well, but I had a few suggestions for improvements going a bit beyond the scope of the original issue:

1. In addition to the free-text search, specific fields can now be searched with queries like `resources:hepdata_lib` or `data_abstract:"CERN-LHC"`.  These examples could be given in the "Advanced Search Tips" ([`search_help.html`](https://github.com/HEPData/hepdata/blob/add-additional-desc-to-osindex/hepdata/modules/search/templates/hepdata_search/modals/search_help.html)).
2. The original issue #557 mentions that the goal is "these additional resources become searchable and findable in automatic queries".  In practice, this means that enough information should be given in the JSON format of the search results and record pages to enable (e.g.) a Python script to download the resource files.   Currently, the `resources` in the JSON format contains only a list of `description` fields of the resources.  It should be expanded to give also a `url` and `type` so that `resources` is now a list of dictionaries (`description`, `url`, `type`).  The `type` can be set to `file_type` of the `DataResource` object, while `url` is the `file_location` if it starts with `http`, otherwise (for a resource file not an external link) construct the `url` from the `id` similar to [this line](https://github.com/HEPData/hepdata/blob/a07723f0639a9ba2e0339716ea0c2f5a93db187d/hepdata/ext/opensearch/document_enhancers.py#L110).
3. The `resources` of an individual data table should be added to the `data_tables` when viewing the JSON for a whole record and also to the JSON for an individual data table.  This is not currently done.
4. You didn't modify the [mapping](https://opensearch.org/docs/latest/field-types/) in [`record_mapping.py`](https://github.com/HEPData/hepdata/blob/add-additional-desc-to-osindex/hepdata/ext/opensearch/config/record_mapping.py), so the new fields `data_abstract` and `resources` take default values (check http://localhost:9200/hepdata-main/_mapping ).  This seems to work with the current code, but it might be necessary to give an explicit mapping after addressing point 2.

ItIsJordan committed:
Fix get json table data url

Fixes and issue where the get_table_details route was not properly routing when load_all flag is not set. This resulted in failure to find the correct route.

Also removes method, as it likely does not matter.

ItIsJordan committed:
Add resource url generation function

Adds a function to generate/return the correct web link for a DataResource object. Adds associated test.

ItIsJordan committed:
Update record_mapping to include resources/data_abstract

Update record_mapping to include resources/data_abstract at the record-level.

ItIsJordan committed:
Add function to generate DataResource dictionary

Adds function to generate a DataResource dictionary data from a DataResource object.
Also adds testing.

ItIsJordan committed:
Add full resource data to OpenSearch index

Adds the full resource data to the OpenSeach index for both records and table resource objects.

ItIsJordan committed:
Add full resource data to table output

Adds full resource data to the output of the process_data_tables function so the full resource data shows on an individual record's JSON output

ItIsJordan committed:
Update search_test.py

Add search testing for new search functionality

ItIsJordan committed:
Update advanced search help

Updates the "Advanced" search help tooltip in search_help.html to include new data_abstract searching, and searching by resource (type, description, url)

ItIsJordan committed:
Merge branch 'main' into add-additional-desc-to-osindex

ItIsJordan committed:
Change use of SITE_URL

Changes use of SITE_URL in the tests to retrieve from app.config.get as this is safer.

ItIsJordan committed:
Add shorthand for resource description searching

Adds a shorthand for "resource.description" searching as "resources" in the OpenSearch query builder

ItIsJordan committed:
Update search help text

Updates search help text in search_help.html to standardise the text for data abstract searching. Also updated to mention "resources" shorthand for description searching.

ItIsJordan committed:
Fix broken test

Add "app" fixture to broken tests that now seem to require it. (where app is used to get site url)

ItIsJordan committed:
Add "resources" data to data table json output

Adds resource informaton to the table_contents in get_table_details, which will be seen in the json output of an individual data table.

ItIsJordan committed:
Update site_url use

Replaces more use of SITE_URL importing directly from the config, which can be unsafe, to use the current_app value provided by flask.

(adding missing imports)

ItIsJordan committed:
Update search_help.html

Updates the search help widget in search_help.html to remove reference to non-shorthand "resources.description". Now just references "resources" as the shorthand term for this.

GraemeWatt reviewed (approved):
This is great!  Thanks a lot.  I'll merge now.
